pkg/types/types.go

```diff
@@ -77,10 +77,17 @@ func ParseRekorLeaf(r io.Reader) (*RekorLeaf, error) {
 		return nil, err
 	}
 
+	if err := l.ParseKeys(); err != nil {
+		return nil, err
+	}
+	return &l, nil
+}
+
+func (l *RekorLeaf) ParseKeys() error {
```

Comment:
Since you're also "validating" the SHA, perhaps name it something like "ValidateLeaf"?